connection/src/error.rs

```diff
@@ -31,8 +30,8 @@ pub enum Error {
     Cipher(CipherError),
     /// Attestation failure: {0}
     Attestation(Box<dyn AttestationError + 'static>),
-    /// Transaction validation failure: {0}
-    TransactionValidation(TransactionValidationError),
+    /// Transaction validation failure: {0:?}: {1}
+    TransactionValidation(ProposeTxResult, String),
```

Comment:
This is the part of this that is a breaking change to the `ThickClient`, since, instead of returning the previous error type, it now returns this error code as a message. But, this is what the server actually gave you over grpc, instead of what you sort of hackishly reconstructed from the grpc response.

For instance, previously if the Validation error was "Ledger", you would just get the empty string, because it's payload is a string on the server side and that can't be marshalled into a proto enum. But now you would actually get the error string on the client side.